docs/docs/adding-search-with-js-search.md

```diff
@@ -1,8 +1,357 @@
+     */
+    termFrequency === true
+      ? (dataToSearch.searchIndex = new JsSearch.TfIdfSearchIndex("isbn"))
+      : (dataToSearch.searchIndex = new JsSearch.UnorderedSearchIndex())
+
+    /**
+     * checks the values in the state and indexes the data revieved based on it
+     */
+    if (indexByTitle) {
+      dataToSearch.addIndex("title")
+    }
+    if (indexByAuthor) {
+      dataToSearch.addIndex("author")
+    }
+
+    dataToSearch.addDocuments(bookList) // adds the data
+    this.setState({ search: dataToSearch })
+  }
+
+  /**
+   * es6 fat arrow function to handle the input change and perfom a search with js-search
+   * in which the results will be added to the state
+   */
+  searchData = e => {
+    const { search } = this.state
+    const queryResult = search.search(e.target.value)
+    this.setState({ searchQuery: e.target.value, searchResults: queryResult })
+  }
+  render() {
+    const { bookList, searchResults, searchQuery, indexByAuthor } = this.state
+    return (
+      <div>
+        <div>
+          <form onSubmit={this.handleSubmit}>
+            <div className="form-group">
+              <label htmlFor="Search" className="form-label">
+                Enter your search here
+              </label>
+              <input
+                id="Search"
+                value={searchQuery}
+                onChange={this.searchData}
+                placeholder="Enter your search here"
+                className="searchQuery"
+              />
+            </div>
+            .....
+          </form>
+          <div>
+            Number of items:
+            {searchQuery === "" ? books.length : searchResults.length}
+            <DataTable data={searchQuery === "" ? books : searchResults} />
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+export default Search
+```
+
+Breaking down the code into smaller parts:
+
+1. When the component is mounted, the `componentDidMount()` lifecycle method is triggered and the data will be fetched.
+2. If no errors occur, the data received is added to the state and the `rebuildIndex()` function is invoked.
+3. The search engine is then created and configured with the options provided in the component's state.
+4. The data that was received is then added and indexed using js-search.
+5. When the contents of the input change, `js-search` starts the search process based on the updated input and returns the contents, which is then presented to the user via the `DataTable` component.
+
+Note:
+
+For brevity purposes, most of the component implementation is omited, with the exception of the `searchData()` function, as this one is responsible for making the search, also the implementation of the `DataTable` component and the page holding this component is not shown. The full code of this example is available [in the js-search repo](https://github.com/gatsbyjs/gatsby/tree/master/examples/using-js-search).
```

Comment:
I like how you've highlighted the relevant parts of the code in the tutorial here. But I can see this part being confusing for someone that's followed along so far, I got a bit stuck working out which files I needed to copy from the example. 

An approach that could make this clearer is to reduce the amount of files used in the example site. So you could do something like this:

- delete `search.css` and apply those styles inline in the DataTable.js component
- Move the code from DataTable.js into SearchContainer.js, and then delete the DataTable.js file

Which might not be strictly best practice, but we can add comments explaining how you'd do it in a "real" site, and it'll make the follow up instructions shorter. 

Doing this means that someone following this tutorial only needs to copy `components/SearchContainer.js` and then edit `pages/index.js` to get their version into a working state.

Then the instructions here can clearly describe which files should be copied from the example site to your local site. After making the previous changes, this could be something like _To get this running in your site, copy the SearchContainer.js file to your site, and then update the file `pages/index.js [as shown here](#)`_

Edit: I added some more thoughts about tightening up this section in the overall PR comment.